tests/Product/Repository/ProductTest.php

```diff
@@ -2,30 +2,69 @@
 
 namespace KickAss\Commerce\Product\Repository;
 
+use KickAss\Commerce\Product\Map\Product;
 use PHPUnit\Framework\TestCase;
 use KickAss\Commerce\Product\Repository\Product as ProductRepository;
 use KickAss\Moltin\Bridge\Moltin\Product as MoltinProduct;
 use \Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
 
-/**
- * @covers ProductRepository
- */
 final class RepositoryProductTest extends TestCase
 {
+    private $moltinMock;
+
     /**
-     * test mapping
+     * @var ProductRepository
      */
-    public function testPopulateProduct()
+    private $product;
+
+    protected function setUp()
     {
-        $product = new ProductRepository(
-            new MoltinProduct(),
+        $this->moltinMock = $this->getMockBuilder(MoltinProduct::class)->getMock();
+        $this->moltinMock->method('getProductItem')
+            ->willReturn(['sku' => 'sample-sku']);
+        $this->moltinMock->method('getProductItemByAttribute')
+            ->willReturn(['sku' => 'sample-sku']);
+        $this->moltinMock->method('getProductList')
+            ->willReturn([
+                ['sku' => 'sample-sku'],
+                ['sku' => 'second-sample']
+            ]);
+
+        $this->product = new ProductRepository(
+            $this->moltinMock,
             new ObjectNormalizer()
         );
+    }
+
+    public function testLoad()
+    {
+        $map = $this->product->load(123);
+
+        $this->assertEquals('sample-sku', $map->getSku());
+        $this->assertInstanceOf(Product::class, $map);
+    }
+
+    public function testLoadByAttribute()
+    {
+        $map = $this->product->loadByAttribute(
+            'slug',
+            'sample-slug'
+        );
 
-        $map = $product->populateProductRepository([
-            'sku' => 'foobar'
-        ]);
+        $this->assertEquals('sample-sku', $map->getSku());
+        $this->assertInstanceOf(Product::class, $map);
+    }
+
+    public function testSearch()
+    {
+        $results = $this->product->search(['sku' => 'value']);
 
-        $this->assertEquals('foobar', $map->getSku());
+        $this->assertInternalType('array', $results);
+        $this->assertEquals(2, count($results));
+        $first = $results[0];
+        $this->assertEquals('sample-sku', $first->getSku());
```

Comment:
this feels weird to me and I really wanted to loop through the array and check each sku but I was too lazy to think about a "cool" way to do that